map2loop/thickness_calculator.py

```diff
@@ -340,3 +347,208 @@ def compute(
+
+    Attributes:
+        thickness_calculator_label (str): A string that stores the label of the thickness calculator.
+        For this class, it is "ThicknessCalculatorGamma".
+
+    Methods:
+        compute(units: pandas.DataFrame, stratigraphic_order: list, basal_contacts: pandas.DataFrame, map_data: MapData)
+        -> pandas.DataFrame: Calculates the thickness in meters for each unit in the stratigraphic column.
+
+    '''
+
+    def __init__(self):
+        self.sorter_label = "ThicknessCalculatorGamma"
+
+    @beartype.beartype
+    def compute(
+        self,
+        units: pandas.DataFrame,
+        stratigraphic_order: list,
+        basal_contacts: geopandas.GeoDataFrame,
+        structure_data: pandas.DataFrame,
+        map_data: MapData,
+    ) -> pandas.DataFrame:
+        """
+        Method overview:
+        - define perpendicular line, with strike perpendicular to the stratigraphic measurement's strike
+        - find intersection points between the perpendicular line and the geological contacts
+        - Perform the following checks:
+            1) is there more than one intersection?
+            2) are the intersections between two different lithologies, and if so, grab the neighbouring lithologies only.
+            3) is the distance between the two intersections less than half the map dimensions? (avoids incorrect intersections to be picked)
+            4) is the stratigraphic measurement strike within 30 degrees of the strike of the geological contacts?
+
+        - once the intersections pass the checks, calculate the thickness of the unit at the intersection points, using the general formula L*sin(dip)
+
+        Attributes:
+        -----------
+            units (pandas.DataFrame): the data frame of units to add thicknesses to
+            stratigraphic_order (list): a list of unit names sorted from youngest to oldest
+            basal_contacts (geopandas.GeoDataFrame): basal contact geo data with locations and unit names of
+            the contacts (columns must contain ["ID","basal_unit","type","geometry"])
+            structure_data (pandas.DataFrame): sampled structural data
+            map_data (map2loop.MapData): a catchall so that access to all map data is available
+
+        Returns:
+        --------
+            pandas.DataFrame: units dataframe with added thickness columns:
+            "gammaThickness" is the median thickness of the unit,
+            "gammaStdDev" is the standard deviation of the thickness of the unit
+
+        Note:
+        -----
+            This method is highly dependent on the existence of stratigraphic measurements that follow the strike of the geological contacts.
+            If an unit does not contain a stratigraphic measurement, the thickness will not be calculated. Future versions of map2loop will attempt to solve for this.
```

Comment:
You can suggest Thickness calculator beta can be a remedy to this problem because it interpolates orientation measurements, but it has its limits as well.

Reply:
In https://github.com/Loop3D/map2loop/pull/59/commits/bfebedccb0802b56a54c60c9889ddc3f6802f7e4